api/config/v1alpha1/metric_types.go

```diff
@@ -10,6 +10,13 @@ type ProxyMetrics struct {
 	Prometheus *PrometheusProvider `json:"prometheus,omitempty"`
 	// Sinks defines the metric sinks where metrics are sent to.
 	Sinks []MetricSink `json:"sinks,omitempty"`
+	// ProxyStatMatcher defines configuration for reporting custom Envoy stats.
```

Comment:
* what is the default behavior,  is it all stats ? if so can we add a note around that ?
* what happens when the user specifies a match string that doesnt match any metric ?
* should we combine `ProxyStatMatcher` & `HistogramBucketSettings` into one setting ?
    * for a specific `match` of a stat, use this bucket, if bucket unspecified, use default bucket setting 
 * `proxyStatsMatcher` seems too wordy, thoughts on `matches` or `statsSelectors`    
 * can we use the existing style of `Type` used in many places in this project and upstream for `InclusionPrefixes` etc e.g.
 ```
 matches:
 - type: "Prefix"
    value: "cluster_upstream_rq"
 ```

Reply:
- The current default behavior is to output metrics prefixed with `cluster_manager, listener_manager, server, cluster.xds-grpc`, I think these are the most basic metrics. Default behaviors need to be discussed，releated  https://github.com/envoyproxy/envoy/issues/28179
- The current metric generation behavior is to only generate metrics matching default rules and matching custom rules. If no custom rules are matched, only the default behavioral metrics are generated.
- ProxyStatMatcher & HistogramBucketSettings can be used to reduce memory and CPU overhead from envoy stats system. Maybe combine them into `customStats` field.
-  For [StatsMatcher](https://www.envoyproxy.io/docs/envoy/latest/api-v3/config/metrics/v3/stats.proto#config-metrics-v3-statsmatcher) ，we actually only used the inclusion_list (No stats are enabled except for those matching one of the supplied StringMatcher). It may be clearer to use InclusionXX here

Reply:
@AliceProxy @kflynn and I discussed this in the meeting today are here are some thoughts
* lets create a separate issue for Histogram buckets and better understand the type of buckets users wants to configure
* the current API is not very intuitive
```
metrics:
  customMetricScale:
    statsMatcher:
      inclusionMatches:
      - type:  ...
        value: ...

```

there seem be multiple layers of `match` which feels cumbersome

Instead can we stick to the earlier suggestion
```
metrics:
  matches:
  - type: ...
     value: ...
```

and lets document which metrics/stats are computed when
* when `matches` is `nil`
* when `matches` is not empty

Reply:
create a issue for this: https://github.com/envoyproxy/gateway/issues/1786

Reply:
> @AliceProxy @kflynn and I discussed this in the meeting today are here are some thoughts
> 
> * lets create a separate issue for Histogram buckets and better understand the type of buckets users wants to configure
> * the current API is not very intuitive
> 
> ```
> metrics:
>   customMetricScale:
>     statsMatcher:
>       inclusionMatches:
>       - type:  ...
>         value: ...
> ```
> 
> there seem be multiple layers of `match` which feels cumbersome
> 
> Instead can we stick to the earlier suggestion
> 
> ```
> metrics:
>   matches:
>   - type: ...
>      value: ...
> ```
> 
> and lets document which metrics/stats are computed when
> 
> * when `matches` is `nil`
> * when `matches` is not empty

done,using `matches` directly seems convenient to use.

- When Matches is nil(not set), gateway by default create and expose only a subset of Envoy stats with prefixes(cluster_manager,listener_manager,server,cluster.xds-grpc) expressions match on the name of stats.

- When Matches is not empty, this option is to control creation of additional Envoy stats with prefix, suffix, and regex expressions match on the name of the stats.